### PR TITLE
[CI] Add `kubectl kustomize` run to check our manifests definition

### PR DESCRIPTION
**Pull request by jstourac**
Added a simple check to CI to run `kubectl kustomize` so we are sure that our kustomization.yml files don't fail with an error. This should avoid issues like this in the future hopefully red-hat-data-services/notebooks#253.

This is tested automatically by this PR.

This should be backported to our downstream repo.

https://issues.redhat.com/browse/RHOAIENG-7886

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: .github/workflows/code-quality.yaml [hunk 1]
```diff
@@ -51,3 +51,13 @@ jobs:
           echo "Starting Hadolint"
           find . -name "Dockerfile" | xargs ./hadolint --config ./ci/hadolint-config.yaml
           echo "Hadolint done"
+
+      # This simply checks that the manifests and respective kustomization.yaml finishes without an error.
+      - name: Check kustomize manifest
+        id: kustomize-manifests
+        run: |
+          type kubectl || echo TODO
```
Comment (jiridanek) [inline .github/workflows/code-quality.yaml:59]: `set -x`, to avoid having to write these echos?

**Review by jiridanek — APPROVED** [review]

**Comment by jiridanek** [comment]
This job runs

```yaml
---
name: Code static analysis
on: [pull_request] # yamllint disable-line rule:truthy
```

I think that it should be push and pull_request, so that it's possible to test it on personal fork, before creating a PR. Wdyt?

**Review by jiridanek — APPROVED** [review]

File: .github/workflows/code-quality.yaml (hunk 1 above)
Comment (jstourac) [reply .github/workflows/code-quality.yaml]: maybe in the future - I added also extra lines with `----` separators for better text separation visually

**Comment by jstourac** [comment]
> This job runs
> 
> ```yaml
> ---
> name: Code static analysis
> on: [pull_request] # yamllint disable-line rule:truthy
> ```
> 
> I think that it should be push and pull_request, so that it's possible to test it on personal fork, before creating a PR. Wdyt?

Probably we can, I'm not sure what else this change will cause. But I'm not gonna do it in this PR :slightly_smiling_face:

**Comment by jiridanek** [comment]
> > This job runs
> > ```yaml
> > ---
> > name: Code static analysis
> > on: [pull_request] # yamllint disable-line rule:truthy
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I think that it should be push and pull_request, so that it's possible to test it on personal fork, before creating a PR. Wdyt?
> 
> Probably we can, I'm not sure what else this change will cause. But I'm not gonna do it in this PR 🙂

* https://github.com/opendatahub-io/notebooks/pull/547

File: .github/workflows/code-quality.yaml (hunk 1 above)
Comment (jiridanek) [reply .github/workflows/code-quality.yaml]: banners; meh

image

;P

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve

:+1:

**Pull request merged**